### PR TITLE
fix(mechanics): Don't launch with player when not landed on the same planet

### LINKED ISSUE #9510: Escorts moved via mini-map missing from shipyard
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Move selected escorts via a galaxy map/mini-map right-click, to system A (with shipyard).  Main fleet then lands on B.  Using the player/ship info panel, **_park_** the remote escorts on A.  Fly main fleet from B to A and land. 

Now, when you visit the shipyard/outfitters, the initially moved escorts do not appear in the ships list.  If you **_unpark_** them, then **depart and immediately land**, they are now visible in the shipyard/outfitters.

### Steps to Reproduce

1. Go to '_Beta Lupi'_ and capture pirates (which become 'Escorts X').
2. Use galaxy/mini-map and right-click to send escorts X to '_Seginus/Clark'_
3. Jump fleet to '_Men'_.  Wait for escorts X to reach '_Seginus'_.  Land on '_Thule'_. 
4. On Player/Ship Info screens, park escorts X at '_Seginus'_.
5. Depart 'Thule', jump to '_Seginus'_ and land on '_Clark'_.
6. In Shipyard/Outfitter screens escorts X are **not shown**.
7. In Player/Ship Info, unpark escorts X.
8. In Shipyard/Outfitter screens escorts X are **still not shown**.
9. Depart and immediately land again.
10. Escorts X are now visible in Shipyard and outfitters

### Expected Behavior

Ships remotely sent to other systems and parked should be immediately visible in Shipyard/Outfitters when main fleet arrives.

### Screenshots

_No response_

### Link to save file

_No response_

### Operating System

Linux Mint 20.3 Una

### Game Source

GitHub Releases

### Game Version

v 0.10.4 appimage downloaded 23/10/23

### Additional Information

Issue seen on laptop and desktop machines.  Same OS/game version on both.

### PR DESCRIPTION
**Pull request by TomGoodIdea**
**Bugfix:** This PR fixes #9510

## Fix Details
Changed the logic of ship placement on takeoff, so that ships parked in space don't launch with player. Also, ships now get recharged on takeoff from the planet they're landed on, which may be different from that of the player.

## Testing Done
Tested that this fixes the linked issue.

### CODE REVIEW

File: source/Engine.cpp [hunk 1]
```diff
@@ -354,16 +354,18 @@ void Engine::Place()
...
 		if(ship->GetSystem() == system && !ship->IsDisabled())
 		{
-			const Personality &person = ship->GetPersonality();
+			const Personality &personality = ship->GetPersonality();
 			bool hasOwnPlanet = ship->GetPlanet();
-			bool launchesWithPlayer = (planet && planet->CanLand(*ship))
-					&& !person.IsStaying() && !person.IsWaiting()
-					&& (!hasOwnPlanet || (ship->IsYours() && ship->GetPlanet() == planet));
+			bool launchesWithPlayer = (ship->GetPlanet() == planet
+					&& !personality.IsStaying() && !personality.IsWaiting());
```
Comment (quyykk) [inline source/Engine.cpp:360]: Doesn't this break NPCs without planets that should be launching with the player? I think it should be enough to restrict `hasOwnPlanet` for non-player ships.

File: source/Engine.cpp [hunk 2]
```diff
@@ -354,16 +354,18 @@ void Engine::Place()
...
 			const StellarObject *object = hasOwnPlanet ?
 					ship->GetSystem()->FindStellar(ship->GetPlanet()) : nullptr;
 			// Default to the player's planet in the case of data definition errors.
-			if(person.IsLaunching() || launchesWithPlayer || (hasOwnPlanet && !object))
+			if(personality.IsLaunching() || launchesWithPlayer || (hasOwnPlanet && !object))
 			{
+				if(hasOwnPlanet)
```
Comment (quyykk) [inline source/Engine.cpp:366]: Isn't this going to trigger for the other cases as well, say for a player escort that landed with the player on a planet?

File: source/Engine.cpp (hunk 1 above)
Comment (TomGoodIdea) [reply source/Engine.cpp:360]: you mean NPCs with `launching` personality?

Comment (quyykk) [reply source/Engine.cpp:360]: No, NPCs without launching, without staying and without waiting.

**Comment by TomGoodIdea** [comment]
In the linked issue we agreed that the problem is that ships appear to be landed when they aren't, not that ships parked in other systems don't land automatically.

**Comment by warp-core** [comment]
I can't see any good reason why an escort parked in the current system but not "landed" on the current planet shouldn't just be assigned to this planet if it can land here. Even if it were parked on a different planet, if it can land where I am, just make it available here? This feels like realism at the expense of gameplay. Needing to micromanage the planets my escorts are on doesn't seem very desirable.

**Comment by oldpenguin5** [comment]
please see my comment in https://github.com/endless-sky/endless-sky/issues/9510#issuecomment-2380850229

As [warp-core](https://github.com/warp-core) commented [on Mar 7](https://github.com/endless-sky/endless-sky/pull/9547#issuecomment-1983610575), this fix does not address the problem in #9510

**Pull request closed without merging**